llama_index/indices/struct_store/retriever.py

```diff
@@ -0,0 +1,63 @@
+import logging
+from typing import Any, List, Optional, Union
+
+from sqlalchemy import Table
+
+from llama_index.data_structs.node import NodeWithScore
+from llama_index.indices.base_retriever import BaseRetriever
+from llama_index.indices.query.schema import QueryBundle
+from llama_index.indices.struct_store.sql_table import SQLTableIndex
+
+DEFAULT_QUERY_TMPL = (
+    "Please return the relevant tables (including the full schema) "
+    "for the following query: {orig_query_str}"
+)
+
+logger = logging.getLogger(__name__)
+
+
+class SQLTableRetriever(BaseRetriever):
+    """Retriever for SQL Table Index."""
+
+    def __init__(
+        self,
+        index: SQLTableIndex,
+        query_tmpl: Optional[str] = None,
+        tables: Optional[List[Union[str, Table]]] = None,
+        **kwargs: Any
+    ) -> None:
+        self._index = index
+        self._query_tmpl = query_tmpl or DEFAULT_QUERY_TMPL
+        self._tables = tables
+
+    def _retrieve(
+        self,
+        query_bundle: QueryBundle,
+    ) -> List[NodeWithScore]:
+        """Retrieve nodes."""
+        vector_store_index = self._index.vector_store_index
+
+        # Specific table case. Use initialized tables if specified.
+        if self._tables:
+            vector_index_retriever = vector_store_index.as_retriever(similarity_top_k=1)
```

Comment:
why is similarity_top_k=1 hardcoded

Reply:
also see comments below, not sure why this needs to be vector store specific

Reply:
Doesn't have to be a vector store actually.

Reply:
why is that the case?

Reply:
Ah you mean including extra context info. Hmm..

Reply:
Fixed